apps/hadoop.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apps
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
+)
+
+type MetricsReceiverHadoop struct {
+	confgenerator.ConfigComponent          `yaml:",inline"`
+	confgenerator.MetricsReceiverShared    `yaml:",inline"`
+	confgenerator.MetricsReceiverSharedJVM `yaml:",inline"`
+
+	CollectJVMMetics *bool `yaml:"collect_jvm_metrics"`
```

Comment:
Now that the Elasticsearch PR is merged, can we make use of the common util here as well? https://github.com/GoogleCloudPlatform/ops-agent/blame/c145e38a9640f55eb46761398ad5446ecc9cea07/confgenerator/config.go#L479

Since there is a planned follow up PR for Hadoop, and this change is not user visible, I'm okay with doing it in the follow up PR as well.

Reply:
Updated!

Reply:
Thanks!